src/browser/components/tabbrowser/content/tabbrowser-js.patch

```diff
@@ -103,3 +121,12 @@ index ef857bd81f2cd7c163ecc74ac1cf81a0b63ce838..807eb0c493f15643412b05d8dad81d36
          aTab.selected ||
          aTab.closing ||
          // Tabs that are sharing the screen, microphone or camera cannot be hidden.
+@@ -7870,7 +7911,7 @@ var TabContextMenu = {
+     );
+     contextUnpinSelectedTabs.hidden =
+       !this.contextTab.pinned || !multiselectionContext;
+-
++    gZenPinnedTabManager.updatePinnedTabContextMenu(this.contextTab);
```

Comment:
This could be moved into the pinned tab manager, instead of having a patch.

Something like

```
popupElement.addEventListener('popupshowing', <update popup>);
```